tls/extensions/s2n_key_share.h

```diff
@@ -25,4 +25,5 @@
 #define S2N_SIZE_OF_NAMED_GROUP             2
 #define S2N_SIZE_OF_KEY_SHARE_SIZE          2
 
-extern int s2n_ecdhe_parameters_send(struct s2n_ecc_evp_params *ecc_evp_params, struct s2n_stuffer *out);
+S2N_RESULT s2n_ecdhe_send_public_key(struct s2n_stuffer *out, struct s2n_ecc_evp_params *ecc_evp_params);
+int s2n_ecdhe_parameters_send(struct s2n_ecc_evp_params *ecc_evp_params, struct s2n_stuffer *out);
```

Comment:
`s2n_ecdhe_parameters_send` already existed (and is referenced in many tests) and includes writing the curve id. The new `s2n_ecdhe_send_public_key` method doesn't write the curve id, and was added to avoid duplicating the logic to generate/write the classic keyshare from the hybrid PQ code: https://github.com/lrstewart/s2n/blob/keyshare_refactor_2/tls/extensions/s2n_client_key_share.c#L119-L125 The arguments are swapped to match the other method the PQ code uses,`s2n_kem_send_public_key`. Without updating a lot more tests, the argument order of the new `s2n_ecdhe_send_public_key` method can either match `s2n_ecdhe_parameters_send` or `s2n_kem_send_public_key`, but not both. Do you think I should match `s2n_ecdhe_parameters_send` instead?

Reply:
Ooof.

I would probably match `s2n_ecdhe_parameters_send` since that method calls our new method `s2n_ecdhe_send_public_key`. We definitely need comments on the differences between these methods though.